clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -78,23 +80,24 @@
     private final Collection<Node> brokers;
     private final Node controller;
     private final List<TopicMetadata> topicMetadata;
+    private String clusterId = null;
```

Comment:
This should be final. You need an `if/else` in the constructor and assign it to null in case it's not available.